src/types/ISwap.ts

```diff
@@ -32,28 +20,26 @@ export type PoolData = {
     token2: Token;
 };
 
-export type Transaction = {
-    id: string;
-};
-
 export type SwapArgs = {
     tokenIn: Token;
     amountIn: bigint;
     amoundOutMinimum?: bigint;
 };
 
+export type ListPoolToken = Token | kongswap.ListPoolToken;
+
 export interface IPool {
-    swap(args: SwapArgs): Promise<bigint>;
-    quote(args: SwapArgs): Promise<bigint>;
-    getMetadata(): Promise<ICSPoolMetadata>;
+    swap(args: SwapArgs | kongswap.SwapArgs): Promise<bigint | kongswap.SwapResponse>;
+    quote(args: SwapArgs | kongswap.QuoteArgs): Promise<bigint | kongswap.QuoteResponse>;
+    getMetadata(): Promise<ICSPoolMetadata | kongswap.PoolMetadata | null>;
     getPoolData(): PoolData;
     isForToken(token: Token): boolean;
     getTokens(): [Token, Token];
     getLPInfo(): Promise<ICSLPInfo>;
 }
 
 export interface IDex {
-    listTokens(): Promise<Token[]>;
-    listPools(token1?: Token, token2?: Token): Promise<IPool[]>;
-    getPool(token1: Token, token2: Token): Promise<IPool>;
+    listTokens(): Promise<ICSToken[] | kongswap.KongSwapToken[]>;
+    listPools(token1?: ListPoolToken, token2?: ListPoolToken): Promise<IPool[]>;
+    getPool(token1: ListPoolToken, token2: ListPoolToken): Promise<IPool>;
```

Comment:
I don't like that adding a new Dex will always change this interface. Also the ListPoolToken was confusing because you can't see the relation to KongSwapToken right away. How about just using the Token type for the interface? The implementations for the DEXes can return the wider type.

Reply:
can I change to this?
```
export type Token = {
    symbol?: string;
    name?: string;
    address: string;
     chain?: string;
};
```